test/sim/misc/inversebattle.js

```diff
@@ -79,6 +79,15 @@ describe('Inverse Battle', function () {
 		assert.equal(battle.p2.active[0].hp, battle.p2.active[0].maxhp);
 	});
 
+	it('should not affect move-based immunities', function () {
+		battle.setPlayer('p1', {team: [{species: "Hariyama", ability: 'guts', moves: ['earthquake']}]});
+		battle.setPlayer('p2', {team: [{species: "Klefki", ability: 'prankster', moves: ['magnetrise']}]});
+		battle.makeChoices('move earthquake', 'move magnetrise');
+		battle.makeChoices('move earthquake', 'move magnetrise');
+		assert(battle.log[battle.lastMoveLine + 1].startsWith('|-immune|'));
+		assert.equal(battle.p2.active[0].hp, battle.p2.active[0].maxhp);
+	});
+
```

Comment:
```suggestion
	it('should not affect move-based immunities', function () {
		battle.setPlayer('p1', {team: [{species: "Hariyama", moves: ['earthquake']}]});
		battle.setPlayer('p2', {team: [{species: "Klefki", ability: 'prankster', moves: ['magnetrise']}]});
		battle.makeChoices();
		assert(battle.log[battle.lastMoveLine + 1].startsWith('|-immune|'));
		assert.fullHP(battle.p2.active[0]);
	});

```
Should simplify things a bit.

Reply:
Why don't you fix up all of the other tests while you're there?

Reply:
Sure, I can do that.